rcl/include/rcl/timer.h

```diff
@@ -157,7 +158,8 @@ rcl_timer_init(
   rcl_context_t * context,
   int64_t period,
   const rcl_timer_callback_t callback,
-  rcl_allocator_t allocator);
+  rcl_allocator_t allocator,
+  bool autostart);
```

Comment:
I think that we want to add a second method with the old API and mark it as deprecated so that there's a cycle where we don't break users of the current API.

Reply:
I forgot that in addition to no default arguments you can't overload either in pure C. Nevermind.

Reply:
Yeah, I believe there are other packages using this method. Should we create PR's solving this issue as well before merging this one?

Reply:
Because this is such a core function, we should definitely tick-tock deprecate this one before we change it.

Towards that end, we should make a new API with a similar name, but different enough in the eyes of C.  Some ideas:

* `rcl_timer_init2`
* `rcl_timer_init_autostart`

(I don't love those, but other things along that line would be fine as well).  Then we should mark the old API as deprecated, and then finally we should update all of the in-core users of the old API to the new API.

Reply:
yeah, that is correct, probably we can include API negotiation for that. user can know which feature is enabled, and then using common structure with configuration mode, just call `rcl_timer_configure(COMMAND, common)` to the initialized timer. so that we can avoid this `tick-tock` every time we change the API.

but this is suggestion here, and not directly related to this specific feature. maybe we can consider with different issue 1st.